src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
```

Comment:
It is better to use specific names for the values.
For example 'w' instead of 'p' and 'ch' instead of 'p1'.
It is much easier to read and understand the logic when values have specific names not only for you but for your colleagues as well.
You can check it in all functions.